### ISSUE TITLE
About trainable variables

### ISSUE THREAD

**Issue by arufus** [ifzhang/FairMOT#223]
Hi @ifzhang 
Thanks for the great work!

I have a question about the trainable variable in networks and loss. Can you help me with this?

When reading the code, I found that there are also trainable parameters u in loss (eg. `self.classifier`).

```python
class MotLoss(torch.nn.Module):
    def __init__(self, opt):
        ......
        self.classifier = nn.Linear(self.emb_dim, self.nID)
......

model = create_model(opt.arch, opt.heads, opt.head_conv)
optimizer = torch.optim.Adam(model.parameters(), opt.lr)
......

model = DLASeg('dla{}'.format(num_layers), heads,
                 # pretrained=True,
                 pretrained=False,
                 down_ratio=down_ratio,
                 head_conv=head_conv)
```

The optimizer seems only optimizes the parameters in DLASeg and does not optimize the parameters in MotLoss. Can you tell me the purpose of this?

**Comment by ifzhang** [comment]
I add them to the optimizer now and it can improve the performance of re-ID:
https://github.com/ifzhang/FairMOT/blob/2a353b16a591bf997aadd5d24b73e599672906c4/src/lib/trains/base_trainer.py#L30
